src/core/utils/components/MiddleTextTruncation.tsx

```diff
@@ -0,0 +1,61 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { useOverflow } from '../hooks/useOverflow';
+import { CommonProps } from '../props';
+
+const ELLIPSIS_CHAR = '…';
+
+export type MiddleTextTruncationProps = {
+  /**
+   * Text to truncate.
+   */
+  text: string;
+  /**
+   * Number of characters to leave at the end.
+   * @default 6
+   */
+  endCharsCount?: number;
+} & CommonProps;
+
+/**
+ * Truncates text with the ellipsis in the middle,
+ * leaving defined number of chars at the end.
+ * @example
+ * <MiddleTextTruncation text='Here can be a very long text' />
+ */
+export const MiddleTextTruncation = (props: MiddleTextTruncationProps) => {
+  const { text, endCharsCount = 6, style, ...rest } = props;
+
+  const [ref, visibleCount] = useOverflow(text);
+
+  const truncatedText = React.useMemo(() => {
+    if (visibleCount < text.length) {
+      return `${text.substr(
+        0,
+        visibleCount - endCharsCount - ELLIPSIS_CHAR.length,
+      )}${ELLIPSIS_CHAR}${text.substr(text.length - endCharsCount)}`;
+    } else {
+      return text;
+    }
+  }, [endCharsCount, text, visibleCount]);
+
+  return (
+    <span
+      style={{
+        display: 'flex',
+        minWidth: 0,
+        whiteSpace: 'nowrap',
+        ...style,
+      }}
+      ref={ref}
```

Comment:
In the future we might want to forward the ref but also might want to have an `as` prop. But we can release without it

Reply:
might

Reply:
I am not sure at the end about api and approach, but I guess fir the version first should be ok :?